contrib/format-msgpack/src/main/java/org/apache/drill/exec/store/msgpack/MsgpackReaderContext.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.msgpack;
+
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.hadoop.fs.Path;
+
+import com.fasterxml.jackson.core.JsonParseException;
+
+import io.netty.buffer.DrillBuf;
+import jline.internal.Log;
+
+public class MsgpackReaderContext {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(MsgpackReaderContext.class);
+
+  // Data we're consuming
+  public Path hadoopPath;
+  public long runningRecordCount = 0;
+  public long parseErrorCount;
+  public int recordCount;
+  public DrillBuf workBuf;
+
+  public boolean printToConsole;
+
+  public boolean lenient;
+
+  public boolean readBinaryAsString;
+
+  public boolean useSchema;
+
+  public long currentRecordNumberInFile() {
+    return parseErrorCount + runningRecordCount + recordCount + 1;
+  }
+
+  public void handleAndRaise(String suffix, Exception e) throws UserException {

```

Comment:
Here's where you'd use `UserException` and its builders rather than a custom version.